src/Microsoft.ApiDesignGuidelines.Analyzers/Core/PropertyNamesShouldNotMatchGetMethods.cs

```diff
@@ -6,44 +6,112 @@
+                                                                     DiagnosticCategory.Design,
+                                                                     DiagnosticSeverity.Warning,
+                                                                     isEnabledByDefault: true,
+                                                                     description: s_localizableDescription,
+                                                                     helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182253.aspx",
+                                                                     customTags: WellKnownDiagnosticTags.Telemetry);
+
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(SameTypeRule, BaseTypeRule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(DeclaringTypeRule, BaseTypeRule);
+
+        internal const string s_get = "Get";
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            
+            // Analyze properties, methods 
+            analysisContext.RegisterSymbolAction(symbolContext =>
+            {
+                AnalyzeSymbol(symbolContext.Symbol, symbolContext);
+
+            }, SymbolKind.Property, SymbolKind.Method);
+        }
+
+        private static void AnalyzeSymbol(ISymbol symbol, SymbolAnalysisContext context)
+        {
+            string identifier;
+
+            if (symbol.Kind == SymbolKind.Property)
+            {
+                //if property then target search is to find methods that start with Get and have the property name
+                identifier = s_get + symbol.Name;
+            }
+            else if (symbol.Kind == SymbolKind.Method && symbol.Name.StartsWith(s_get))
+            {
+                //if method starts with Get then target search is to find properties that have the method name sans Get
+                identifier = symbol.Name.Substring(3);
+            }
+            else
+            {
+                //if method name doesn't start with Get exit
+                return;
+            }
+
+            //boolean variable used to exit out of the inner and outer for loops
+            var matchFound = false;
+
+            //get the collection of the containing type plus all the derived types
+            var types = symbol.ContainingType.GetBaseTypesAndThis();
+
+
+            foreach (var type in types)
+            {
+                var membersFound = type.GetMembers(identifier);
+                if (membersFound != null && membersFound.Length > 0)
+                {
+                    foreach (var member in membersFound)
+                    {
+                        if ((symbol.Kind == SymbolKind.Property && member.Kind == SymbolKind.Method) ||
+                            (symbol.Kind == SymbolKind.Method && member.Kind == SymbolKind.Property && symbol.ContainingType != type))
```

Comment:
Discussed off-line. Without that check, diagnostic will fire twice for the same issue